### PR TITLE
build(deps): upgrade eslinting

### LINKED ISSUE #161: React Hooks Eslint Plugin
**Is your feature request related to a problem? Please describe.**
While we are using React hooks in our app, it is also important for us to enforce the [rules of hooks](https://reactjs.org/docs/hooks-rules.html) while doing so. Since we are now an open-source project, it will be ideal to automate the checks.

**Describe the solution you'd like**
One possible solution is to make use of the [eslint-plugin-react-hooks](https://www.npmjs.com/package/eslint-plugin-react-hooks) ESLint plugin, which adds React hooks linting into our existing ESLint lint runs. Additionally, this plugin has first-party support from the React team.

### LINKED ISSUE #208: Update ESLint dependencies
### Overview, Problem Description, Summary, or Abstract

Dependencies relating to ESLint - including ESLint itself, configs and plugins -  should be updated to the latest versions in a single PR, with checks to ensure that linting continues to work for the core engineering team. The PR should also resolve #161 by adding the appropriate ESLint plugin.

### Context or Background

As part of efforts to enable automatic updates to our dependencies, we need to first do a manual one-off update and verify that GoGovSG still works. This will be done in phases, starting with all dependencies relating to linting.

### Test Plan

Ensure changes to code via linting, if any, are within reason and documented

### PR DESCRIPTION
**Pull request by stanleynguyen**
- Upgraded eslint & related plugins/configs to latest
- Exclusion of import/no-cycle as we have a cycle client/actions/login <> client/actions/user

Resolves #208 
Resolves #161

### CODE REVIEW

File: src/client/components/HomePage/LandingGraphicSliver.jsx [hunk 1]
```diff
@@ -103,9 +103,9 @@ const useStyles = makeStyles((theme) =>
 const LandingGraphicSliver = () => {
   const classes = useStyles()
   const topPaddingMultipler = () => {
-    const theme = useTheme()
-    const isMediumWidth = useMediaQuery(theme.breakpoints.up('md'))
-    const isDesktopWidth = useMediaQuery(theme.breakpoints.up('lg'))
+    const theme = UseTheme()
+    const isMediumWidth = UseMediaQuery(theme.breakpoints.up('md'))
+    const isDesktopWidth = UseMediaQuery(theme.breakpoints.up('lg'))
```
Comment (LoneRifle) [inline src/client/components/HomePage/LandingGraphicSliver.jsx:108]: `useTheme` and `useMediaQuery` really are hooks and should hence retain camelCase naming. However, `topPaddingMultipler()` is also meant to be a hook, and should hence follow hook convention.

I would like you to reinstate the camelCasing for `useTheme` and `useMediaQuery`, and rename `topMaddingMultipler()` to `useTopPaddingMultiplier()` (note the typo)

File: src/client/components/HomePage/LandingGraphicSliver.jsx [hunk 2]
```diff
@@ -102,10 +102,10 @@ const useStyles = makeStyles((theme) =>
...
 const LandingGraphicSliver = () => {
   const classes = useStyles()
-  const topPaddingMultipler = () => {
-    const theme = useTheme()
-    const isMediumWidth = useMediaQuery(theme.breakpoints.up('md'))
-    const isDesktopWidth = useMediaQuery(theme.breakpoints.up('lg'))
+  const useTopPaddingMultipler = () => {
+    const theme = useTheme() // eslint-disable-line react-hooks/rules-of-hooks
+    const isMediumWidth = useMediaQuery(theme.breakpoints.up('md')) // eslint-disable-line react-hooks/rules-of-hooks
+    const isDesktopWidth = useMediaQuery(theme.breakpoints.up('lg')) // eslint-disable-line react-hooks/rules-of-hooks
```
Comment (LoneRifle) [inline src/client/components/HomePage/LandingGraphicSliver.jsx:108]: Given that the hooks are being used within a custom hook, `eslint-disable-line react-hooks/rules-of-hooks` should be unnecessary
```suggestion
    const theme = useTheme()
    const isMediumWidth = useMediaQuery(theme.breakpoints.up('md'))
    const isDesktopWidth = useMediaQuery(theme.breakpoints.up('lg'))
```

Comment (stanleynguyen) [reply src/client/components/HomePage/LandingGraphicSliver.jsx:108]: Oh right right @LoneRifle

**Comment by LoneRifle** [comment]
Hey @stanleynguyen - as the PR indicates, you're still rebasing the branch on top of `master`, not `develop`. You might have to either do it again, or cherry-pick the individual commits out of the branch onto `develop`. Sorry for the inconvenience!

**Comment by LoneRifle** [comment]
The display of this PR seems to be out of sync with the actual git tree, so I'm going to close this one and recreate it

**Pull request closed without merging**